packages/main/src/RPA/application.py

```diff
@@ -0,0 +1,167 @@
+"""Base utilities for COM applications like Word, Excel, Outlook."""
+
+import atexit
+import logging
+import platform
+import struct
+from contextlib import contextmanager
+from pathlib import Path
+
+if platform.system() == "Windows":
+    import win32api
+    import win32com.client
+    from pywintypes import com_error as COMError
+    from win32com.client import constants  # pylint: disable=unused-import
+else:
+    logging.getLogger(__name__).warning(
+        "Any `RPA.*.Application` library works only on Windows platform!"
+    )
+    # As these are imported anyway from here and should be defined on non-Windows OSes.
+    COMError = Exception
+    constants = None
+
+
+def _to_unsigned(val):
+    return struct.unpack("L", struct.pack("l", val))[0]
+
+
+def to_path(path: str) -> Path:
+    return Path(path).expanduser().resolve()
+
+
+def to_str_path(path: str) -> str:
+    return str(to_path(path))
+
+
+# TODO(cmin764; 16 Aug 2023): Create a `handles_com_error` decorator if we get a little
+#  too repetitive with context managers guarding. (to decorate the keywords)
+@contextmanager
+def catch_com_error():
+    """Try to convert COM errors to human-readable format."""
+    try:
+        yield
+    except COMError as err:  # pylint: disable=no-member
+        if err.excepinfo:
+            try:
+                msg = win32api.FormatMessage(_to_unsigned(err.excepinfo[5]))
+            except Exception:  # pylint: disable=broad-except
+                msg = err.excepinfo[2]
+        else:
+            try:
+                msg = win32api.FormatMessage(_to_unsigned(err.hresult))
+            except Exception:  # pylint: disable=broad-except
+                msg = err.strerror
+        raise RuntimeError(msg) from err
+
+
+class MetaApplication(type):
+    """Metaclass enhancing every final class inheriting from a base using this."""
+
+    def __new__(cls, *args, **kwargs) -> type:
+        final_class = super().__new__(cls, *args, **kwargs)
+        bases = final_class.__mro__[1:]
+        if len(bases) < 2:  # not enhancing bases
+            return final_class
+
+        super_class = bases[0]
+        final_class.__doc__ += super_class.__doc__
+        if final_class.APP_DISPATCH is None:
+            raise ValueError(
+                "An `APP_DISPATCH` has to be defined in this"
+                f" {final_class.__name__!r} class"
+            )
+        return final_class
+
+
+class BaseApplication(metaclass=MetaApplication):
```

Comment:
More on [metaclasses](https://docs.python.org/3/reference/datamodel.html#metaclasses). (in case it gets fuzzy)